From b8cc72d7bc74255d1fece8b543c0228a18d3c7b3 Mon Sep 17 00:00:00 2001
From: montellese <montellese@kodi.tv>
Date: Thu, 16 Jul 2020 22:27:33 +0200
Subject: [PATCH] [videodb] improve (unique) indices on actor_link table after
 bffa7bb5e33aff1325e090eddbba16a2740637bb.

Due to the missing INDEX on actor_link (media_id, media_type(20), actor_id)
performance of queries which JOIN the actor_link table against a media item
table on media_id and media_type drastically decreased.
---
 xbmc/video/VideoDatabase.cpp | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/xbmc/video/VideoDatabase.cpp b/xbmc/video/VideoDatabase.cpp
index 68c7241fc761..10e4afa22fb2 100644
--- a/xbmc/video/VideoDatabase.cpp
+++ b/xbmc/video/VideoDatabase.cpp
@@ -265,9 +265,11 @@ void CVideoDatabase::CreateAnalytics()
   m_pDS->exec("CREATE INDEX ix_uniqueid2 ON uniqueid(media_type(20), value(20))");
 
   m_pDS->exec("CREATE UNIQUE INDEX ix_actor_1 ON actor (name(255))");
-  m_pDS->exec("CREATE INDEX ix_actor_link_1 ON actor_link (media_type(20))");
-  m_pDS->exec("CREATE UNIQUE INDEX ix_actor_link_2 ON "
-              "actor_link (actor_id, media_id, media_type(20), role(255))");
+  m_pDS->exec("CREATE UNIQUE INDEX ix_actor_link_1 ON "
+              "actor_link (actor_id, media_type(20), media_id, role(255))");
+  m_pDS->exec("CREATE INDEX ix_actor_link_2 ON "
+              "actor_link (media_id, media_type(20), actor_id)");
+  m_pDS->exec("CREATE INDEX ix_actor_link_3 ON actor_link (media_type(20))");
 
   CreateLinkIndex("tag");
   CreateForeignLinkIndex("director", "actor");
@@ -5670,7 +5672,7 @@ void CVideoDatabase::UpdateTables(int iVersion)
 
 int CVideoDatabase::GetSchemaVersion() const
 {
-  return 117;
+  return 118;
 }
 
 bool CVideoDatabase::LookupByFolders(const std::string &path, bool shows)
